src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookingFacadeTest.kt

```diff
@@ -445,6 +446,78 @@ class BookingFacadeTest {
       }
     }
 
+    @Test
+    fun `should send events and emails on cancellation of court booking by prison user`() {
+      setupCourtPrimaryContactsFor(PRISON_USER)
+
+      val prisoner = Prisoner(courtBooking.prisoner(), MOORLAND, "Bob", "Builder", yesterday())
+
+      whenever(cancelVideoBookingService.cancel(1, PRISON_USER)) doReturn courtBooking.apply { cancel(PRISON_USER) }
+      whenever(prisonerSearchClient.getPrisoner(courtBooking.prisoner())) doReturn prisoner
+      whenever(emailService.send(any<CancelledCourtBookingUserEmail>())) doReturn Result.success(emailNotificationId to "user template id")
+      whenever(emailService.send(any<CancelledCourtBookingCourtEmail>())) doReturn Result.success(emailNotificationId to "court template id")
+
+      facade.cancel(1, PRISON_USER)
+
+      inOrder(cancelVideoBookingService, outboundEventsService, emailService, notificationRepository) {
+        verify(cancelVideoBookingService).cancel(1, PRISON_USER)
```

Comment:
This is good - verifying the order in which things occurred. Note to self - do more of this.

Reply:
Yes, I keep meaning to do this but don't always remember.